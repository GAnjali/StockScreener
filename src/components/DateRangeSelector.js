import React, {Component} from "react";
import {DateRangePicker} from "react-dates";
import 'react-dates/initialize';
import 'react-dates/lib/css/_datepicker.css';
import "../styles/react_dates_overrides.css"
import "../styles/styles.css";

class DateRangeSelector extends Component {
    state = {
        focusedInput: null
    };

    handleDateChange = ({startDate, endDate}) => {
        this.props.handleSelectDate(startDate, endDate);
    };

    handleFocusChange = focusedInput => this.setState({focusedInput});

    render() {
        return (
            <DateRangePicker
                endDate={this.props.endDate}
                endDateId="endDate"
                focusedInput={this.state.focusedInput}
                isOutsideRange={() => null}
                onDatesChange={this.handleDateChange}
                onFocusChange={this.handleFocusChange}
                startDate={this.props.startDate}
                startDateId="startDate"
            />
        )
    }
}

export default DateRangeSelector;