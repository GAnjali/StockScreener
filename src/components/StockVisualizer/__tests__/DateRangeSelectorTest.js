import {shallow} from "enzyme";
import React from "react";
import DateRangeSelector from "../DateRangeSelector";

describe("DateRangeSelectorTests", () => {
    it("Should render without crashing", () => {
        shallow(<DateRangeSelector/>);
    });

    it('should match the DateRangeSelector snapshot', function () {
        const dashboardComponent = shallow(<DateRangeSelector/>);
        expect(dashboardComponent).toMatchSnapshot();
    });
});