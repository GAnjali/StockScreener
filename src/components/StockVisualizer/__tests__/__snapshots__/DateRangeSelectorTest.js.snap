// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRangeSelectorTests should match the DateRangeSelector snapshot 1`] = `ShallowWrapper {}`;
